### ISSUE TITLE
BOM downloads default to "SNAPSHOT" version when project version is unassigned

### ISSUE THREAD

**Issue by lukas-braune** [DependencyTrack/dependency-track#4112]
### Current Behavior

When downloading BOMs via Dependency-Track's frontend (Components > Download BOM) and no version is assigned to the project (Project Details > Version), the generated CycloneDX document defaults to the version `SNAPSHOT` in `metadata.component.version`, e.g.:

```
{
  "bomFormat" : "CycloneDX",
  "specVersion" : "1.5",
  "serialNumber" : "urn:uuid:c60df495-ab6a-4677-b825-5940811a7c3b",
  "version" : 1,
  "metadata" : {
    "timestamp" : "2024-09-02T08:10:01Z",
    "tools" : [
      {
        "vendor" : "OWASP",
        "name" : "Dependency-Track",
        "version" : "4.11.7"
      }
    ],
    "component" : {
      "type" : "application",
      "bom-ref" : "f6b74dfe-e8aa-41c9-ab8b-03e285165273",
      "name" : "Test BOM",
      "version" : "SNAPSHOT"
    }
  },
```

### Steps to Reproduce

1. Create a new project and leave the version field blank.
2. In the newly created project, go to _Components_ tab, click on _Download BOM_, and select one of the available options.
3. Check the generated BOM for `metadata.component.version` to confirm that the version is set to `SNAPSHOT`.

### Expected Behavior

If no version is assigned to a project in Dependency-Track, the `metadata.component.version` field in the downloaded BOMs should be left empty. This field is optional in the CycloneDX specification.

### Dependency-Track Version

4.11.7

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

15.8

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
The version is set here: https://github.com/DependencyTrack/dependency-track/blob/e672e6b474c13a452516b6ebd9ecae563d9dd361/src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java#L710-L714

It seems like `component.version` was a required field in earlier versions of the CycloneDX spec.

**Comment by nscuro** [comment]
We could technically exploit an oversight in the spec by setting it to an empty string instead: https://github.com/CycloneDX/specification/issues/461

**Comment by Gepardgame** [comment]
@nscuro I added a fix for that [#4142](https://github.com/DependencyTrack/dependency-track/pull/4142)
